src/backend/libpq/rangerrest.c

```diff
@@ -86,132 +98,132 @@ RangerACLResult parse_ranger_response(char* buffer)
-                {
-                    first = pstrdup(name);
-                }
-                else if (idx == 1)
-                {
-                    second = pstrdup(name);
-                }
-                else
-                {
-                    third = pstrdup(name);
-                }
-            }
-
-            if (first != NULL)
-            {
-                json_object *jfirst = json_object_new_string(first);
-                json_object_object_add(jresource, "database", jfirst);
-            }
-            if (second != NULL)
-            {
-                json_object *jsecond = json_object_new_string(second);
-                json_object_object_add(jresource,
-                        (kind == ACL_KIND_LANGUAGE) ? "language" : "schema", jsecond);
-            }
-            if (third != NULL)
-            {
-                json_object *jthird = json_object_new_string(third);
-                json_object_object_add(jresource,
-                         (kind == ACL_KIND_CLASS) ? "table" :
-                         (kind == ACL_KIND_SEQUENCE) ? "sequence" : "function", jthird);
-            }
-
-            if (first != NULL)
-                pfree(first);
-            if (second != NULL)
-                pfree(second);
-            if (third != NULL)
-                pfree(third);
-            break;
-        }
-        case ACL_KIND_OPER:
-        case ACL_KIND_CONVERSION:
-        case ACL_KIND_DATABASE:
-        case ACL_KIND_TABLESPACE:
-        case ACL_KIND_TYPE:
-        case ACL_KIND_FILESYSTEM:
-        case ACL_KIND_FDW:
-        case ACL_KIND_FOREIGN_SERVER:
-        case ACL_KIND_EXTPROTOCOL:
-        {
-            json_object *jobject = json_object_new_string(object);
-            json_object_object_add(jresource, AclObjectKindStr[kind], jobject);
-            break;
-        }
-        default:
-            elog(ERROR, "unrecognized objkind: %d", (int) kind);
-    } // switch
-
-    json_object_object_add(jelement, "resource", jresource);
-    
-    //ListCell *cell;
-    //foreach(cell, arg_ptr->actions)
-    //{
-      char tmp[7] = "select";
-      json_object* jaction = json_object_new_string((char *)tmp);
-      //json_object* jaction = json_object_new_string((char *)cell->data.ptr_value);
-      json_object_array_add(jactions, jaction);
-    //}
-    json_object_object_add(jelement, "privileges", jactions);
-    json_object_array_add(jaccess, jelement);
-
-  } // foreach
-
-  json_object_object_add(jrequest, "user", juser);
-  json_object_object_add(jrequest, "access", jaccess);
-
-  json_object *jreqid = json_object_new_string("1");
-  json_object_object_add(jrequest, "requestId", jreqid);
-  json_object *jclientip = json_object_new_string("123.0.0.21");
-  json_object_object_add(jrequest, "clientIp", jclientip);
-  json_object *jcontext = json_object_new_string("SELECT * FROM DDDDDDD");
-  json_object_object_add(jrequest, "context", jcontext);
-
-  return jrequest;
+	json_object *juser = NULL;
```

Comment:
Here I think you'd better set the tab/space again to make the diff minimal for review. If it is tricky to do that, let it be.